cmd/influxd/run.go

```diff
@@ -89,6 +90,35 @@ func execRun(args []string) {
 			go func() { log.Fatal(http.ListenAndServe(config.ApiHTTPListenAddr(), sh)) }()
 		}
 		log.Printf("DataNode#%d running on %s", s.ID(), config.ApiHTTPListenAddr())
+
+		// Spin up any Graphite servers
+		for _, c := range config.Graphites {
+			if !c.Enabled {
+				continue
+			}
+
+			// Configure Graphite parsing.
+			parser := graphite.NewParser()
+			parser.Separator = c.NameSeparatorString()
+			parser.LastEnabled = (c.NamePosition == "last")
+
+			// Start the relevant server.
+			if strings.ToLower(c.Protocol) == "tcp" {
+				g := graphite.NewTCPServer(parser, s)
+				g.Database = c.Database
+				err := g.ListenAndServe(c.ConnectionString(config.BindAddress))
+				if err != nil {
+					log.Println("failed to start TCP Graphite Server", err.Error())
+				}
+			} else {
+				g := graphite.NewUDPServer(parser, s)
+				g.Database = c.Database
+				err := g.ListenAndServe(c.ConnectionString(config.BindAddress))
+				if err != nil {
+					log.Println("failed to start UDP Graphite Server", err.Error())
+				}
+			}
```

Comment:
Can you change this so it checks for `"udp"` explicitly? Then have a final `else` block if it doesn't match `tcp` or `udp`.